src/WebJobs.Script/ExtensionBundle/ExtensionBundleManager.cs

```diff
@@ -261,9 +261,9 @@ internal static string FindBestVersionMatch(VersionRange versionRange, IEnumerab
                     version = versionRange.Satisfies(version) ? version : null;
                 }
                 return version;
-            }).Where(v => v != null);
+            }).Where(v => v != null).OrderByDescending(version => version.Version).ToList();
 
-            var matchingVersion = bundleVersions.OrderByDescending(version => version.Version).FirstOrDefault();
+            var matchingVersion = ResolvePlatformReleaseChannelVersion(bundleVersions);
```

Comment:
There are many null-handlers for `matchingVersion` below via `?.` operators. But I wonder what the actual impact of this being null is? Is this an error scenario? If so, should this be updated to log and maybe error out if `matchingVersion == null`?